src/main/java/eu/bbmri_eric/negotiator/database/model/Resource.java

```diff
@@ -71,6 +67,32 @@ public class Resource {
   @Exclude
   private AccessForm accessForm;
 
+  @ManyToMany(mappedBy = "resources")
+  @Builder.Default
+  @Setter(AccessLevel.NONE)
+  private Set<Network> networks = new HashSet<>();
+
+  public void addNetwork(Network network) {
+    this.networks.add(network);
+    if (!network.getResources().contains(this)) {
+      network.addResource(this);
+    }
+  }
+
+  public Set<Network> getNetworks() {
+    if (Objects.isNull(this.networks)) {
+      return Set.of();
+    }
+    return Collections.unmodifiableSet(this.networks);
+  }
+
+  public void removeNetwork(Network network) {
+    this.networks.remove(network);
+    if (network.getResources().contains(this)) {
+      network.removeResource(this);
```

Comment:
I do not think it makes sense to have this API on both the Resource and the Network. Wouldn't it be better to just have it in one of them?